pkg/process/metadata/parser/java/spring.go

```diff
@@ -0,0 +1,205 @@
+			parts[pl-1] = "BOOT-INF/classes/" + parts[pl-1]
+			isClasspath = true
+		}
+
+		doAppend := func(name string, profile string) {
+			name = filepath.Clean(name)
+			if isClasspath {
+				classpaths[profile] = append(classpaths[profile], filepath.ToSlash(name))
+			} else {
+				files[profile] = append(files[profile], filepath.ToSlash(abs(name, cwd)))
+			}
+		}
+		if strings.HasSuffix(parts[pl-1], "/") {
+			// we have a path: add all the possible filenames
+			tmp := parts[pl-1] + name
+			// there is an extension based priority also: first properties then yaml
+			for _, profile := range profiles {
+				tmp2 := tmp + "-" + profile
+				for _, ext := range []string{".properties", ".yaml", ".yml"} {
+					doAppend(tmp2+ext, profile)
+				}
+			}
+			for _, ext := range []string{".properties", ".yaml", ".yml"} {
+				doAppend(tmp+ext, "")
+			}
+		} else {
+			// just add it since it's a direct file
+			doAppend(parts[pl-1], "")
+		}
+	}
+	return files, classpaths
+}
+
+// abs returns the absolute path resolved from cws if not already absolute
+// it slightly differs from the golang built in version because on windows the built-in considers absolute a path only
+// if the volume name is specified. This one only requires a path to begin with os.PathSeparator to be absolute
+func abs(path string, cwd string) string {
+	// on windows IsAbs likely returns false when the drive is missing
+	// hence, since we accept also paths, we test if the first char is a path separator
+	if !(filepath.IsAbs(path) || path[0] == os.PathSeparator) && len(cwd) > 0 {
+		return filepath.Join(cwd, path)
+	}
+	return path
+}
+
+// newSpringBootArchiveSourceFromReader return a PropertyGetter combined source with properties sources from the jar application.
+func newSpringBootArchiveSourceFromReader(reader *zip.Reader, patternMap map[string][]string) map[string]*props.Combined {
+	ret := make(map[string]*props.Combined)
+	// probably we need something more efficient
+	matcher := antpath.New()
+	for _, f := range reader.File {
+		name := f.Name
+		// the generalized approach implies visiting also jar in BOOT-INF/lib but here we skip it
+		// to minimize the scanning time given that the general habit is to package config
+		// directly into the application and not in a lib embedded into the app
+		if strings.HasPrefix(name, "BOOT-INF/classes/") {
+			for profile, patterns := range patternMap {
+				for _, pattern := range patterns {
+					if matcher.Match(pattern, name) {
+						rc, err := f.Open()
+						if err != nil {
+							log.Trace("Unable to open", name, ":", err)
+						} else {
+							source, err := func() (*props.PropertyGetter, error) {
+								defer rc.Close()
+								source, err := newPropertySourceFromStream(rc, f.Name, f.UncompressedSize64)
+								return source, err
+							}()
+							if err != nil {
+								log.Trace("Error creating property source", err)
+							} else {
+								log.Trace("Loaded", f.Name)
```

Comment:
is that useful log?
trace logs, and every log in hot-code-path has a negative impact on performance

Reply:
removed logs